### PR TITLE
GIF Player: add on change props to listen to gif play/pause

### PR DESCRIPTION
**Pull request by AquiGorka**
@benwiley4000 hello again, another PR here:

This changes introduce an onChange prop to listen to changes in the gif state (play/pause).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -93,6 +93,8 @@ class GifPlayerContainer extends React.Component {
   toggle () {
     this.setState({
       playing: !this.state.playing
+    }, () => {
+      this.props.onChange && this.props.onChange(this.state.playing)
```
Comment (benwiley4000) [inline src/index.js]: A few comments:
1. I prefer the form:
```js
if (typeof this.props.onChange === 'function') {
  this.props.onChange(...);
}
```
2. I also prefer handling this in `componentDidUpdate` - we can just check if the playing state is different than the last.
3. Semicolon would be great. 😉

**Review by benwiley4000 — CHANGES_REQUESTED** [review]
Great idea! Thanks for submitting this. 😊

I've got a few comments.. after those are resolved we can ship it.

**Comment by benwiley4000** [comment]
Also I'm thinking about it and we could probably me more descriptive than "onChange". Maybe "onTogglePlay" or something? In case we have to add other handlers later on.

**Comment by AquiGorka** [comment]
>I prefer the form:
```javascript
if (typeof this.props.onChange === 'function') {
  this.props.onChange(...);
}
```
Added `destructuring` with default value, let me know if this is alright with you.

>I also prefer handling this in `componentDidUpdate` - we can just check if the playing state is different than the last.

Done.

>Semicolon would be great. 😉

;)

File: src/index.js [hunk 2]
```diff
@@ -69,6 +69,10 @@ class GifPlayerContainer extends React.Component {
 
   componentDidUpdate (prevProps, prevState) {
     this.updateImages(prevState);
+    const { onTogglePlay=()=>{} } = this.props;
```
Comment (benwiley4000) [inline src/index.js]: I guess it's more slick/compact this way but I strongly prefer checking if `typeof onTogglePlay === 'function'` to creating a new arrow function unnecessarily.

**Review by benwiley4000 — COMMENTED** [review]
Awesome, thanks for the changes! One more thing.

File: src/index.js (hunk 2 above)
Comment (AquiGorka) [reply src/index.js]: I did ask which you preferred because I felt you'd like to check explicitly–though the other way is, as you mention, more "slick" ;)

Comment (benwiley4000) [reply src/index.js]: I do have a soft spot in my heart for those slick answers, but I generally find the other way easier to read and a bit more performant. Thanks! 🙂

Comment (benwiley4000) [reply src/index.js]: I do have a soft spot in my heart for those slick answers, but I generally find the other way easier to read and a bit more performant. Thanks! 🙂

**Review by benwiley4000 — APPROVED** [review]

**Comment by AquiGorka** [comment]
@benwiley4000 will you let me know when you update on `npm`? I have an app eager to use this update ;)

**Comment by benwiley4000** [comment]
@AquiGorka yep it's out! v0.4.0. 🙂

**Pull request merged**